docs/querying/datasource.md

```diff
@@ -22,43 +22,317 @@ title: "Datasources"
+|`joinType`|`INNER` or `LEFT`.|
+
+#### Join performance
+
+Joins are a feature that can significantly affect performance of your queries. Some performance tips and notes:
+
+1. Joins are especially useful with [lookup datasources](#lookup), but in most cases, the
+[`LOOKUP` function](sql.html#string-functions) performs better than a join. Consider using the `LOOKUP` function if
+it is appropriate for your use case.
+2. When using joins in Druid SQL, keep in mind that it can generate subqueries that you did not explicitly include in
+your queries. Refer to the [Druid SQL](sql.md#query-translation) documentation for more details about when this happens
+and how to detect it.
+3. One common reason for implicit subquery generation is if the types of the two halves of an equality do not match.
+For example, since lookup keys are always strings, the condition `druid.d JOIN lookup.l ON d.field = l.field` will
+perform best if `d.field` is a string.
+4. As of Druid {{DRUIDVERSION}}, the join operator must evaluate the condition for each row. In the future, we expect
+to implement both early and deferred condition evaluation, which we expect to improve performance considerably for
+common use cases.
+
+#### Future work for joins
+
+Joins are an area of active development in Druid. The following features are missing today but may appear in
+future versions:
+
+- Preloaded dimension tables that are wider than lookups (i.e. supporting more than a single key and single value).
+- RIGHT OUTER and FULL OUTER joins. Currently, they are partially implemented. Queries will run but results will not
+always be correct.
+- Performance-related optimizations as mentioned in the [previous section](#join-performance).
+- Join algorithms other than broadcast hash-joins.
+
+### `union`
+
+<!--DOCUSAURUS_CODE_TABS-->
+<!--Native-->
 ```json
 {
-       "type": "union",
-       "dataSources": ["<string_value1>", "<string_value2>", "<string_value3>", ... ]
+  "queryType": "scan",
+  "dataSource": {
+    "type": "union",
+    "dataSources": ["<dataSourceName1>", "<dataSourceName2>", "<dataSourceName3>"]
+  },
+  "columns": ["column1", "column2"],
+  "intervals": ["0000/3000"]
 }
 ```
+<!--END_DOCUSAURUS_CODE_TABS-->
 
-Note that the data sources being unioned should have the same schema.
-Union Queries should be always sent to a Broker/Router process and are *NOT* supported directly by the Historical processes.
+Union datasources allow you to treat two or more table datasources as a single datasource. The datasources being unioned
+do not need to have identical schemas. If they do not fully match up, then columns that exist in one table but not
+another will be treated as if they contained all null values in the tables where they do not exist.
```

Comment:
Are there any performance notable implications of using union datasources?

Reply:
It is indirectly described [here](https://github.com/apache/druid/pull/9704/files#diff-203b8f56b37fff44bf7ab3b38b94308bR105-R107).

Reply:
yes, they're discussed in query-execution. I'll add a link.